File: Server/Core/DTOs/NewSaintDto.cs
using System.ComponentModel.DataAnnotations;

public class NewSaintDto
{
    [Required]
    public string Name { get; set; } = default!;

    [Required]
    public string Country { get; set; } = default!;

    [Range(1, 21)]
    public int Century { get; set; }

    [Required]
    public string Image { get; set; } = default!;

    [Required]
    public string Description { get; set; } = default!;

    [Required]
    public string MarkdownContent { get; set; } = default!;

    public string? Title { get; set; }

    public DateOnly? FeastDay { get; set; }

    public string? PatronOf { get; set; }

    public int? ReligiousOrderId { get; set; }

    [MaxLength(5)]
    public List<int>? TagIds { get; set; }

    public string? Slug { get; set; }
}

File: Client/src/app/features/saints/interfaces/new-saint-dto.ts
export interface NewSaintDto {
  name: string;
  country: string;
  century: number;
  image: string;
  description: string;
  markdownContent: string;
  title?: string | null;
  feastDay?: string | null; 
  patronOf?: string | null;
  religiousOrderId?: number | null;
  tags?: string[]; 
}
